src/mpi/comm/comm_split_type.c

```diff
@@ -393,6 +393,38 @@ static int compare_info_args(const void *sendbuf, void *recvbuf, int count, MPIR
     goto fn_exit;
 }
 
+static int compare_info_hint(const char *hintval, MPIR_Comm * comm_ptr, int *info_args_are_equal)
+{
+    int hintval_size = strlen(hintval);
+    int hintval_size_global;
+    char *hintval_global = NULL;
+    int mpi_errno = MPI_SUCCESS;
+
+    /* Compare hintval string length */
+    mpi_errno =
+        reduce_info_hint(&hintval_size, &hintval_size_global, sizeof(int) / sizeof(char), comm_ptr,
```

Comment:
The implementation of `reduce_info_hint` still seems a bit overly complex, but I think we can solve that later.

Reply:
Do we need the `reduce_info_hint` function?  Can't that just be embedded into this function since this is the only place that it's used?